content/blog/2025-05-dev-update/index.md

```diff
@@ -0,0 +1,100 @@
+  He also worked on usability, submitting [a PR to TLC](https://github.com/tlaplus/tlaplus/pull/1180) to display values of any ∀-bound names when an invariant is violated.
+  Finally, he submitted [a PR for the TLC interpreter](https://github.com/tlaplus/tlaplus/pull/1178) simplifying the `isEmpty()` method on set datatypes.
+- [Karolis Petrauskas](https://github.com/kape1395) worked on the [TLAPM-based](https://github.com/tlaplus/tlapm) TLA⁺ language server, changing [TLAPM's info output stream](https://github.com/tlaplus/tlapm/pull/209) and further refining [proof step renaming](https://github.com/tlaplus/tlapm/pull/211).
+  He also improved the language server's integration with the TLA⁺ VS Code Extension ([1](https://github.com/tlaplus/vscode-tlaplus/pull/375), [2](https://github.com/tlaplus/vscode-tlaplus/pull/376)).
+- Longtime TLA⁺ tools maintainer [Markus Kuppe](https://github.com/lemmy/) had quite an active month, adding [company-level execution statistics to TLC](https://github.com/tlaplus/tlaplus/issues/1170), multiple user-friendly warnings ([1](https://github.com/tlaplus/tlaplus/pull/1173), [2](https://github.com/tlaplus/tlaplus/pull/1174), [3](https://github.com/tlaplus/tlaplus/pull/1185)), improved error messages when [TLC encounters non-enumerable values](https://github.com/tlaplus/tlaplus/pull/1179), and variable inclusion in [TLC coverage statistics](https://github.com/tlaplus/tlaplus/pull/1183).
+  He also began writing [a new tool documentation set](https://github.com/tlaplus/tlaplus/pull/1181).
+  For the TLA⁺ VS Code Extension, he updated it to [incorporate level checking information from SANY](https://github.com/tlaplus/vscode-tlaplus/pull/367) and [exposed some APIs](https://github.com/tlaplus/vscode-tlaplus/pull/372) to be used by LLM agents.
+
+### TLA⁺ Foundation-funded updates
+
+Finally, things [Andrew Helwer](https://ahelwer.ca/) (author of this post) worked on - all funded by the TLA⁺ Foundation!
+- I spent a lot of time expanding the [*Create your own TLA⁺ tools*](https://docs.tlapl.us/creating:start) tutorial series, trying to encode as much knowledge about parsing TLA⁺ as possible.
+  As a highlight, the tutorial now has a module on parsing vertically-aligned conjunction & disjunction lists.
+  This is nearly complete; I've finished writing the code for two of the final three modules (operator parameters, actions, then safety checking) and now need to write them up.
+- I created & delivered [a talk](https://www.youtube.com/watch?v=KrhZebeRn90) at the TLA⁺ Community Event summarizing the current state of TLA⁺ development, what the TLA⁺ Foundation has been funding to improve it, and what I want to see in the near future.
+
+### Newbie Corner
```

Comment:
A second topic for the Newbie Corner could be https://github.com/tlaplus/vscode-tlaplus/issues/377, which involves contributing to the VSCode TLA+ extension. This codebase is likely more approachable for newcomers and offers a practical starting point for getting involved.

Reply:
Nice, I'll save that to put in the June newsletter.